app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt

```diff
@@ -0,0 +1,181 @@
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.Truss
+import org.simple.clinic.util.Unicode
+import org.simple.clinic.util.UtcClock
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.hideKeyboard
+import org.threeten.bp.Instant
+import java.util.UUID
+import javax.inject.Inject
+
+class ShortCodeSearchResultScreen(context: Context, attributes: AttributeSet) : RelativeLayout(context, attributes), ShortCodeSearchResultUi {
+
+  @Inject
+  lateinit var uiChangeProducer: ShortCodeSearchResultUiChangeProducer
+
+  @Inject
+  lateinit var utcClock: UtcClock
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  @Inject
+  lateinit var patientRepository: PatientRepository
+
+  @Inject
+  lateinit var facilityRepository: FacilityRepository
+
+  @Inject
+  lateinit var userSession: UserSession
+
+  @Inject
+  lateinit var bloodPressureDao: BloodPressureMeasurement.RoomDao
+
+  @Inject
+  lateinit var schedulersProvider: SchedulersProvider
+
+  private lateinit var binding: ViewControllerBinding<UiEvent, ShortCodeSearchResultState, ShortCodeSearchResultUi>
+
+  private val adapter = GroupAdapter<ViewHolder>()
+
+  private val downstreamEvents = PublishSubject.create<UiEvent>()
+
+  private lateinit var disposable: Disposable
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) {
+      return
+    }
+    hideKeyboard()
+
+    TheActivity.component.inject(this)
+
+    val screenKey = screenRouter.key<ShortCodeSearchResultScreenKey>(this)
+    setupToolBar(screenKey)
+    setupScreen()
+
+    val uiStateProducer = ShortCodeSearchResultStateProducer(
+        shortCode = screenKey.shortCode,
+        patientRepository = patientRepository,
+        userSession = userSession,
+        facilityRepository = facilityRepository,
+        bloodPressureDao = bloodPressureDao,
+        ui = this,
+        schedulersProvider = schedulersProvider
+    )
+    binding = ViewControllerBinding.bindToView(this, uiStateProducer, uiChangeProducer)
+    setupClickEvents()
+  }
+
+  override fun onDetachedFromWindow() {
+    if (::disposable.isInitialized) disposable.dispose()
+    super.onDetachedFromWindow()
+  }
+
+  private fun setupToolBar(screenKey: ShortCodeSearchResultScreenKey) {
+    val shortCode = screenKey.shortCode
+
+    // This is guaranteed to be exactly 7 characters in length.
+    val prefix = shortCode.substring(0, 3)
+    val suffix = shortCode.substring(3)
+
+    val formattedShortCode = "$prefix${Unicode.nonBreakingSpace}$suffix"
+
+    val textSpacingSpan = TextAppearanceWithLetterSpacingSpan(context, R.style.Clinic_V2_TextAppearance_Body0Left_NumericBold_White100)
+
+    toolBar.title = Truss()
+        .pushSpan(textSpacingSpan)
+        .append(formattedShortCode)
+        .popSpan()
+        .build()
+
+    toolBar.setNavigationOnClickListener {
```

Comment:
val popScreenAction = { screenRouter.pop() }

with(toolBar) {
  setNavigationOnClickListener(popScreenAction)
  setOnClickListener(popScreenAction)
}

Something of this sort?